Домашка за 19.08

DenisProkhorov opened the pull request:
- переделал ResponseType
- в респонсе теперь используется ResponseType
- дословно сделал задание с DI: для сервиса и репо сделал интерфейсы и в соответствующих классах (контроллер и сервис соответственно) использую интерфейсы с нужным конструктором

Commit:
Домашка за 19.08

- переделал ResponseType
- в респонсе теперь используется ResponseType
- дословно сделал задание с DI: для сервиса и репо сделал интерфейсы и в соответствующих классах (контроллер и сервис соответственно) использую объекты типа сервис с нужным конструктором

Commit:
Домашка за 23.08

- сделал тесты на 2 оговоренных сценария (когда первый раз делаем реквест и когда произвольный раз делаем реквест)
- удалил неиспользуемый логгер
- Открытый вопрос (надо ли было оставлять DI в связке контроллер/сервис, ибо вроде на занятии такого не было)

vkhizhin commented on app/src/main/java/com/tinkoff/edu/app/LoanCalcController.java:
```diff
@@ -2,14 +2,15 @@
 
 public class LoanCalcController {
+    private LoanCalcLogger loanCalcLogger = new LoanCalcLogger();
+    private LoanCalcService service = new StaticLoanCalcService();
```
Это не DI, ты тут сам создаешь зависимости через оператор new. А DI - это когда они передаются нам через конструктор или сеттеры (лучше делать через конструктор) и мы избавляемся от необходимости вручную создавать объекты-зависимости. Вот в домашке от 23.08 у тебя как раз было сделано как нужно.

Commit:
Домашка за 23.08 - заапплаил комменты:
Пункт 1 (тесты) - действительно забыл включить файл с тестами в коммит
Пункт 2 (тесты) - разнес всё по пакетам, как предлагалось
Пункт 3 (вопрос про DI) - все понял, спасибо.
Пункт 4 (логика) - это как раз следующая домашка (сделаю в отдельном коммите)

Commit:
Домашка за 19.08 - сделал нормальный DI

vkhizhin committed:
Merge branch 'develop' into homework_23

vkhizhin committed:
Merge branch 'homework_23' into develop

vkhizhin commented:
Извини, что поздно проверяю. В общем, все норм. Вмердживаю обе домашки.
